### ISSUE TITLE
Null fields should be ignored

### ISSUE THREAD

**Issue by alvaroarenas** [smallrye/smallrye-graphql#1050]
Hi there,
TL;DR; How can I get SmallRye client to ignore fields when they are null?

I am using the same mutation to create and update an object. When updating an individual field I want to keep the rest unchanged. In that case, the unchanged fields are null. How can I get SmallRye client to omit the null fields in the mutation query?   The client doesn't seem to obey any annotations in the POJO like `@Ignore`

So far the only idea I have come up is to create a second method that will take as input POJO B which is a stripped down version and `@Name` alias for the method name. However, I would prefer not to repeat myself.

Regards,

Alvaro
PS: I used ` @JsonbProperty(nillable = false)` and that one didn't work either.
PS2: I am using smallrye-client v1.3.3

**Comment by jmartisk** [comment]
Hey,

I think this is something that kinda affects the server side as well, because if we just leave out a `null` field from the request; then on the server side, after unmarshalling the object, the Java field will be `null` anyway in both cases, so IIUC the server side will have it a bit harder to tell whether the request contained a null field  (which probably means "set this to null"), or didn't contain it at all (which probably means "don't change this"). The endpoint will have to look into the raw JSON query rather than just work with Java objects. Maybe this can be made easier somehow?! Is there a preferred way to handle this, @phillip-kruger ?


Anyway, if I now move and think for the client side, how should the library know whether a `null` field should be included or not? With the typesafe client (IIUC that's what you're using) it will be hard to set that dynamically (with the dynamic client it's simple, you just don't include it in the request document). 
Maybe the typesafe client could have a boolean configuration property `include-null-fields` that controls this behavior globally? Or a specific annotation like `@IncludeNulls` that could be applied on fields in POJOs, but that's more complicated and I'm not sure if necessary.

**Comment by alvaroarenas** [comment]
Hi,

thx for the answer. Well for starters I infer from your response that this functionality is not there. So, I can continue happily my path towards 2 methods with different POJOs.

I thought that annotating the field with something like `@JsonbProperty(nillable = false)` would do the trick. Or adding a specific annotation like you mention. Maybe we wait for @phillip-kruger 's comment.

**Comment by jmartisk** [comment]
Oh, you may be actually right with `@JsonbProperty`, at first I thought that the `nillable` argument's semantics means it's for validation purposes (rather than serialization), but the JavaDoc says that it turns `null` serialization on/off. In which case we wouldn't need another annotation, just make sure that this one works.

**Comment by phillip-kruger** [comment]
@alvaroarenas -  Can you paste the example code here (for both create and update) ?

**Comment by jmartisk** [comment]
One more observation: 
`nillable` also seems to default to `false`, which means null values should  NOT be included by default, whereas our client code currently DOES include nulls by default. 
It seems that "don't include nulls" should be the correct default, but if we switch to this, it will potentially be a breaking change to some users.
